### PR TITLE
Support pluggable attribute extensions

### PR DESCRIPTION
**Pull request by mirasrael**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -271,6 +271,30 @@ package = Package.new(:dimensions => { 'width' => "2.2", :height => 2, "length"
...
+
+Virtus::Attribute.register_extension(Virtus::Attribute::Format)
+
+class Person
+  include Virtus.model
+
+  attribute :birth_datetime, Datetime, format: '%Y-%m-%d'
+end
+
+Person.attribute_set[:birth_datetime].format
```
Comment (booch) [inline README.md]: We should show that this returns the format string.

File: README.md [hunk 2]
```diff
@@ -271,6 +271,30 @@ package = Package.new(:dimensions => { 'width' => "2.2", :height => 2, "length"
 package.dimensions # => { :width => 2.2, :height => 2.0, :length => 4.5 }
 ```
 
+### Attribute extensions
+
+``` ruby
+module Virtus::Attribute::Format
+  def self.extend?(attribute)
```
Comment (booch) [inline README.md]: The API for this isn't terribly clear. Can we add some docs on what this is for?

**Comment by booch** [comment]
I'm inclined to accept this. But I'd like to see some more documentation explaining how to use it. I'm also curious if there's more to it than just sticking some info in the `attribute_set` and being able to pull it back out.

**Comment by envygeeks** [comment]
Don't forget to rebase your branch.

**Comment by mirasrael** [comment]
not working with Ruby anymore, not going to fix it

**Pull request closed without merging**